### PR TITLE
feat: Add test coverage and example validation to quality gates

### PR DESCRIPTION
**Pull request by evansenter**
## Summary

Adds proactive quality checks for test coverage gaps and example validation:

- **claude-review.md (CI)**: New section 4 for test coverage gap analysis
  - Identify new functions/methods without tests → Important
  - Flag untested code paths → Important  
  - Suggest edge case tests → Suggestion
  
- **Global CLAUDE.md**: Add test coverage and example validation to Quality Gates
  
- **/pr-review local**: Add steps 3a-5a for comprehensive pre-push validation
  - Check for test coverage gaps in changed code
  - Run examples with debug flags (e.g., `LOUD_WIRE=1`) and review output
  - Log to `/tmp/` for inspection

## Motivation

Common end-of-workflow checks that were being asked manually:
1. "Are there any test coverage gaps in this PR?"
2. "Run examples with debug settings and review output"

Now automated in both CI (test coverage analysis) and local review (both checks).

## Test plan
- [ ] Verify claude-review.md changes are picked up by CI on next PR
- [ ] Test `/pr-review local` on a project with examples

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: home/.claude/contrib/prompts/claude-review.md [hunk 1]
```diff
@@ -74,7 +74,24 @@ Before raising any issue, check if it was already addressed in a "Feedback Addre
 
 **Do NOT re-raise issues that appear in Implemented, Skipped, or Deferred sections.**
 
-### 4. Review Criteria
+### 4. Analyze Test Coverage
+
+Before reviewing code quality, analyze test coverage gaps:
```
Comment (evansenter) [inline home/.claude/contrib/prompts/claude-review.md]: test and example coverage gaps

File: home/.claude/contrib/prompts/claude-review.md [hunk 2]
```diff
@@ -89,22 +106,24 @@ Evaluate the code for:
...
    - Violation of project conventions (check CLAUDE.md)
+   - **Missing tests for new public functions/methods**
+   - **Untested new code paths (branches, error handling)**
 
 3. **Suggestions** (nice to have)
    - Code clarity improvements
    - Minor style inconsistencies
    - Documentation gaps
-   - Test coverage opportunities
+   - Test coverage opportunities (edge cases, existing code)
```
Comment (evansenter) [inline home/.claude/contrib/prompts/claude-review.md]: And example opportunities

File: home/.claude/CLAUDE.md [hunk 3]
```diff
@@ -84,6 +84,8 @@ Configured in `~/.claude/settings.json` for autonomous operation:
 
 - Run quality gates appropriate to the project (linter, formatter, tests) before pushing.
 - Significant user-facing features should include examples demonstrating usage - this ensures the codepath doesn't go stale.
+- **Test coverage** - New/changed code should have corresponding tests. Flag missing tests for new public functions/methods.
```
Comment (evansenter) [inline home/.claude/CLAUDE.md]: Also, new user-facing changes should have examples, either new ones or updating existing ones

**Comment by evansenter** [comment]
## Feedback Addressed

### Implemented
- [Suggestion] pr-review.md:3a - Added example coverage analysis alongside test coverage
- [Suggestion] pr-review.md:4a - If no examples but user-facing changes, flag as Important gap
- [Suggestion] claude-review.md:4 - Renamed to "Analyze Test and Example Coverage"
- [Suggestion] claude-review.md:5 - Important issues now include "tests or examples"
- [Suggestion] claude-review.md:5 - Suggestions now include "test/example coverage opportunities"
- [Suggestion] CLAUDE.md - Unified "Test and example coverage" bullet in Quality Gates
